code/modules/hivemind/mobs.dm

```diff
@@ -0,0 +1,913 @@
+						"Hah, did you hear that? They trying to use some kind of we-wep-weapons!",
+						"Useless things, i'm not satisfied.",
+						"This place sucks, man. So creep-p-pew-wepy and no fun, only rudimentary creatures would be live here.")
+
+
+/mob/living/simple_animal/hostile/hivemind/mechiver/Life()
+	. = ..()
+	update_icon()
+
+	//when we have passenger, we torture him
+	if(passenger && prob(15))
+		passenger.apply_damage(rand(5, 10), pick(BRUTE, BURN, TOX))
+		passenger << SPAN_DANGER(pick(
+								"Something grabs your neck!", "You hear whisper: \" It's okay, now you sa-sa-safe! \"",
+								"You've been hit by something metal", "You almost didn't feel your leg!", "Something liquid covers you!",
+								"You feel awful, cadaveric smell", "Something sharp cut your cheek!",
+								"You feel how worm-like thing trying to get inside your head through ear..."))
+		shake_the(src)
+		playsound(src, 'sound/effects/clang.ogg', 70, 1)
+
+
+	//corpse ressurection
+	if(!target_mob && !passenger)
+		for(var/mob/living/Corpse in view(src))
+			if(Corpse.stat == DEAD)
+				if(get_dist(src, Corpse) <= 1)
+					special_ability(Corpse)
+				else
+					walk_to(src, Corpse, 1, 1, 4)
+				break
+
+
+/mob/living/simple_animal/hostile/hivemind/mechiver/speak()
+	if(!client && prob(speak_chance) && speak.len)
+		if(pilot)
+			if(target_mob)
+				visible_message("<b>[name]'s pilot</b> says, [pick(pilot_target_speak)]")
+				say(pick(common_answers))
+			else
+				visible_message("<b>[name]'s pilot</b> says, [pick(pilot_commontalk)]")
+				say(pick(other_answers))
+		else
+			..()
+
+
+//animations
+//updates every life tick
+/mob/living/simple_animal/hostile/hivemind/mechiver/proc/update_icon()
+	if(target_mob && !passenger && (get_dist(target_mob, src) <= 4) && !is_on_cooldown())
+		if(!hatch_closed)
+			return
+		overlays.Cut()
+		if(pilot)
+			flick("mechiver-opening", src)
+			icon_state = "mechiver-chief"
+			overlays += "mechiver-hands"
+		else
+			flick("mechiver-opening_wires", src)
+			icon_state = "mechiver-welcome"
+			overlays += "mechiver-wires"
+		hatch_closed = FALSE
+	else
+		overlays.Cut()
+		hatch_closed = TRUE
+		icon_state = "mechiver-closed"
+		if(passenger)
+			overlays += "mechiver-process"
+
+
+/mob/living/simple_animal/hostile/hivemind/mechiver/AttackingTarget()
+	if(!Adjacent(target_mob))
+		return
+
+	if(world.time > special_ability_cooldown && !passenger)
+		special_ability(target_mob)
+
+	..()
+
+
+//picking up our victim for good 20 seconds of best road trip ever
+/mob/living/simple_animal/hostile/hivemind/mechiver/special_ability(mob/living/target)
+	if(!target_mob && hatch_closed) //when we picking up corpses
+		if(pilot)
+			flick("mechiver-opening", src)
+		else
+			flick("mechiver-opening_wires", src)
+	passenger = target
+	target.loc = src
+	target.canmove = FALSE
+	target << SPAN_DANGER("You've got inside that thing! It's hard to see here, but there something, something that moves around you!")
```

Comment:
You've gotten inside that thing! It's hard to see inside, there's something here, it moves around you!